src/Mobs/Path.h

```diff
@@ -0,0 +1,167 @@
+typedef Vector3<double> Vector3d;
+
+// fwd: cChunkMap.h
+typedef cItemCallback<cChunk> cChunkCallback;
+#endif
+
+/* Various little structs and classes */
+enum class ePathFinderStatus {CALCULATING,  PATH_FOUND,  PATH_NOT_FOUND};
+struct cPathCell;  // Defined inside Path.cpp
+class compareHeuristics
+{
+public:
+	bool operator()(cPathCell * & a_V1,  cPathCell * &  a_V2);
+};
+
+
+
+
+
+class cPath
+#ifndef COMPILING_PATHFIND_DEBUGGER
+: public cChunkCallback
+#endif
+{
+public:
+	/** Creates a pathfinder instance. A Mob will probably need a single pathfinder instance for its entire life.
+	
+	Note that if you have a man-sized mob (1x1x2, zombies, etc), you are advised to call this function without parameters
+	because the declaration might change in later version of the pathFinder, and a parameter-less call always assumes a man-sized mob.
+	
+	If your mob is not man-sized, you are advised to use cPath(width, height), this would be compatible with future versions,
+	but please be aware that as of now those parameters will be ignored and your mob will be assumed to be man sized.
+	
+	@param a_BoundingBoxWidth the character's boundingbox width in blocks. Currently the parameter is ignored and 1 is assumed.
+	@param a_BoundingBoxHeight the character's boundingbox width in blocks. Currently the parameter is ignored and 2 is assumed.
+	@param a_MaxUp the character's max jump height in blocks. Currently the parameter is ignored and 1 is assumed.
+	@param a_MaxDown How far is the character willing to fall? Currently the parameter is ignored and 1 is assumed. */
+	/** Attempts to find a path starting from source to destination.
+	After calling this, you are expected to call Step() once per tick or once per several ticks until it returns true. You should then call getPath() to obtain the path.
+	Calling this before a path is found resets the current path and starts another search.
+	@param a_StartingPoint The function expects this position to be the lowest block the mob is in, a rule of thumb: "The block where the Zombie's knees are at".
+	@param a_EndingPoint "The block where the Zombie's knees want to be".
+	@param a_MaxSteps The maximum steps before giving up. */
+	cPath(
+		cWorld * a_World,
+		const Vector3d & a_StartingPoint, const Vector3d & a_EndingPoint, int a_MaxSteps,
+		double a_BoundingBoxWidth = 1, double a_BoundingBoxHeight = 2,
+		int a_MaxUp = 1, int a_MaxDown = 1
+	);
+	
+	/** Destroys the path and frees its memory. */
+	~cPath();
+	
+	/** Performs part of the path calculation and returns true if the path computation has finished. */
+	ePathFinderStatus Step();
+	
+	/* Point retrieval functions, inlined for performance. */
+	
+	/** Returns the next point in the path. */
+	inline Vector3d GetNextPoint()
+	{
+		ASSERT(m_Status == ePathFinderStatus::PATH_FOUND);
+		return m_PathPoints[m_PointCount - 1 - (++m_CurrentPoint)];
+	}
+	
+	
+	
+	
+	
+	/** Checks whether this is the last point or not. Never call getnextPoint when this is true. */
```

Comment:
Remove the excessive empty lines. 1 empty line between function declarations. 2 empty lines between member variables and functions.